README.md

```diff
@@ -247,35 +295,80 @@ std::optional<lex_item> lexer(std::string_view v) noexcept {
 This support is preliminary, probably the API will be changed.
 
 ```c++
-auto input = "123,456,768"sv;
+#include <ctre.hpp>
+#include <iostream>
+#include <string_veiw>
 
-for (auto match: ctre::range<"([0-9]+),?">(input)) {
+// auto input = "123,456,768";
+extern const char *input;
+
+int main(void) {
+  auto matches = ctre::range<"([0-9]+),?">(input);
+  for (auto match : matches) {
     std::cout << std::string_view{match.get<0>()} << "\n";
+  }
+  return 0;
 }
 ```
+[link to compiler explorer](https://gcc.godbolt.org/z/s4zedb68n)
 
 ### Unicode
 
 ```c++
 #include <ctre-unicode.hpp>
 #include <iostream>
+
 // needed if you want to output to the terminal
 std::string_view cast_from_unicode(std::u8string_view input) noexcept {
-    return std::string_view(reinterpret_cast<const char *>(input.data()), input.size());
+  return std::string_view(reinterpret_cast<const char *>(input.data()), input.size());
 }
-int main()
-{
-    using namespace std::literals;
-    std::u8string_view original = u8"Tu es un génie"sv;
-
-    for (auto match : ctre::range<"\\p{Letter}+">(original))
-        std::cout << cast_from_unicode(match) << std::endl;
-    return 0;
+
+int main() {
+  using namespace std::literals;
+  std::u8string_view original = u8"Tu es un génie"sv;
+
+  for (auto match : ctre::range<"\\p{Letter}+">(original))
+    std::cout << cast_from_unicode(match) << std::endl;
+  return 0;
 }
 ```
 
 [link to compiler explorer](https://godbolt.org/z/erTshe6sz)
 
+## Integration
+You can get [ctre.hpp](https://github.com/hanickadot/compile-time-regular-expressions/blob/main/single-header/ctre.hpp) from the directory `single-header`.
+You need to add:
+
+```C++
+#include "ctre.hpp"
+
+//using namespace ctre;
+```
+This header can be regenerated with `make single-header`.
+### CMake
+If you are using cmake, you can add this directory as subdirectory and link to target `ctre`.
+
+```CMake
+cmake_minimum_required(VERSION 3.8.0)
+
+project(MyProject VERSION 1.0)
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
```

Comment:
CMake 3.8 does not know about C++20, you must not use this value. Please read the [docs](https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html):

> New in version 3.12

This is not the way to set the standard requirement of a target, you must use [compile features](https://cmake.org/cmake/help/latest/manual/cmake-compile-features.7.html#requiring-language-standards) for that.

Reply:
Oh,

> For compilers that have no notion of a standard level, such as Microsoft Visual C++ before 2015 Update 3, this has no effect.